microsoft/testsuites/network/sriov.py

```diff
@@ -542,6 +544,158 @@ def sriov_ethtool_offload_setting(self, environment: Environment) -> None:
+            )
+            for init_interrupts_irq in initial_pci_interrupts_by_irqs:
+                init_irq_number = list(init_interrupts_irq)[0]
+                init_interrupts_value = init_interrupts_irq[init_irq_number]
+                for final_interrupts in final_pci_interrupts_by_irqs:
+                    final_irq_number = list(final_interrupts)[0]
+                    final_interrupts_value = final_interrupts[final_irq_number]
+                    if init_irq_number == final_irq_number:
+                        break
+                # 5. Compare interrupts changes, expected to see interrupts increased.
+                assert_that(final_interrupts_value).described_as(
+                    f"irq {init_irq_number} didn't have an increased interrupts count"
+                    " after iperf3 run!"
+                ).is_greater_than(init_interrupts_value)
+            # 6. Get final interrupts sum per cpu on client node.
+            final_pci_interrupts_by_cpus = Interrupt.get_sum_of_interrupt_data_by_cpus(
+                client_node, client_nic_info.pci_slot
+            )
+            for cpu, init_interrupts_value in initial_pci_interrupts_by_cpus.items():
+                final_interrupts_value = final_pci_interrupts_by_cpus[cpu]
+                # 7. Compare interrupts changes, expected to see interrupts increased.
+                assert_that(final_interrupts_value).described_as(
+                    f"CPU {cpu} didn't have an increased interrupts count"
+                    " after iperf3 run!"
+                ).is_greater_than(init_interrupts_value)
+
+    @TestCaseMetadata(
+        description="""
+        This case is to check whether the network connectivity is lost after running
+         iperf3 for 30 mins.
+
+        Steps,
+        1. Start iperf3 on server node.
+        2. Start iperf3 for 30 minutes on client node.
+        3. Do VF connection test.
+        """,
+        priority=3,
+        timeout=18000,
+        requirement=node_requirement(
+            node=schema.NodeSpace(
+                node_count=2,
+                network_interface=schema.NetworkInterfaceOptionSettings(
+                    nic_count=search_space.IntRange(min=1, max=8),
+                    data_path=schema.NetworkDataPath.Sriov,
+                ),
+            )
+        ),
+    )
+    def sriov_iperf_stress(self, environment: Environment) -> None:
+        server_node = cast(RemoteNode, environment.nodes[0])
+        client_node = cast(RemoteNode, environment.nodes[1])
+        vm_nics = self._initialize_nic_info(environment)
+        client_iperf3_log = "iperfResults.log"
+        server_iperf3 = server_node.tools[Iperf3]
+        client_iperf3 = client_node.tools[Iperf3]
+        # 1. Start iperf3 on server node.
+        server_iperf3.run_as_server()
+
+        # 2. Start iperf3 for 30 minutes on client node.
+        for _, client_nic_info in vm_nics[client_node.name].items():
```

Comment:
Do we need to run on all nics for the perf test case?

Reply:
by default it uses 1 nic?

Reply:
Is it possible to have perf issue on second or other nics? If not, let's test only first nic.

Reply:
not sure the answer, updated into testing against 1 nic

Reply:
Do you mean the test case test only one nic now? But the logic still has the loop on all nics.

Reply:
Use this line to control https://github.com/microsoft/lisa/pull/1637/files#diff-915cefed780f89c28b257620163cf2368234c9a7620b8f8f9ba35cab955daa23R65 only run on primary nic

Reply:
It can be more intuitive. Put only this line in the for loop. If not found, raise an exception. If it's found, run following logic.

Reply:
updated